packages/runtime-vapor/__tests__/for.spec.ts

```diff
@@ -417,20 +417,20 @@ describe('createFor', () => {
       '<li>0. 1</li><li>1. 2</li><li>2. 3</li><li>3. 4</li><!--for-->',
     )
 
-    // change deep value should not update
+    // change
     list.value[0].name = 'a'
     setList()
     await nextTick()
     expect(host.innerHTML).toBe(
-      '<li>0. 1</li><li>1. 2</li><li>2. 3</li><li>3. 4</li><!--for-->',
+      '<li>0. a</li><li>1. 2</li><li>2. 3</li><li>3. 4</li><!--for-->',
     )
 
```

Comment:
This feature was initially removed for performance optimization:
https://github.com/vuejs/vue-vapor/pull/280
https://github.com/vuejs/vue-vapor/commit/b962aa52ff7e71550101ebd77b521928c806ec84